pkg/cmd/openshift-service-serving-cert-signer/config.go

```diff
@@ -0,0 +1,214 @@
+package openshift_service_serving_cert_signer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"path"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/openshift/origin/pkg/cmd/openshift-service-serving-cert-signer/apis/serviceservingcertsigner/v1alpha1"
+)
+
+var (
+	configScheme = runtime.NewScheme()
+	configCodecs = serializer.NewCodecFactory(configScheme)
+)
+
+func init() {
+	v1alpha1.AddToScheme(configScheme)
+}
+
+func readConfig(filename string) (*v1alpha1.ServiceServingCertSignerConfig, error) {
+	config := &v1alpha1.ServiceServingCertSignerConfig{}
+	if err := readYAMLFileInto(filename, config); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+func readAndResolveConfig(filename string) (*v1alpha1.ServiceServingCertSignerConfig, error) {
+	config, err := readConfig(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := resolveConfigPaths(config, path.Dir(filename)); err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+func readYAMLInto(data []byte, obj runtime.Object) error {
+	data, err := kyaml.ToJSON(data)
+	if err != nil {
+		return err
+	}
+	err = runtime.DecodeInto(configCodecs.LegacyCodec(v1alpha1.SchemeGroupVersion), data, obj)
+	return captureSurroundingJSONForError("error reading config: ", data, err)
+}
+
+func readYAMLFileInto(filename string, obj runtime.Object) error {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	err = readYAMLInto(data, obj)
+	if err != nil {
+		return fmt.Errorf("could not load config file %q due to an error: %v", filename, err)
+	}
+	return nil
+}
+
+// TODO: we ultimately want a better decoder for JSON that allows us exact line numbers and better
+// surrounding text description. This should be removed / replaced when that happens.
+func captureSurroundingJSONForError(prefix string, data []byte, err error) error {
```

Comment:
@openshift/api-review this looks suspiciously API helper like.  Is there something better yet?